griptape/tools/rest_api_client/tool.py

```diff
@@ -44,20 +46,27 @@ def full_url(self) -> str:
                 {% if _self.response_body_schema %}The response body must follow this JSON schema: {{ _self.response_body_schema }}{% endif %}
                 """
             ),
-            "schema": Schema({Literal("body", description="The request body."): dict}),
+            "schema": Schema(
+                {
+                    Literal("body", description="The request body."): dict,
+                    schema.Optional(Literal("headers", description="The request headers.")): dict,
+                }
+            ),
         }
     )
     def put(self, params: dict) -> BaseArtifact:
         from requests import put, exceptions
 
-        values = params["values"]
         base_url = self.base_url
-        path = self.path
-        body = values["body"]
+        values = params.get("values", {})
+        path = values.get("path", self.path)
+        body = values.get("body", {})
```

Comment:
Are these changes necessary? We can safely assume these values will be present, so no need for the fallback.